lib/rtsp/session/manager.ex

```diff
@@ -49,7 +64,12 @@ defmodule Membrane.Protocol.RTSP.Session.Manager do
     state = %State{
       transport: transport,
       uri: url,
-      execution_options: options
+      execution_options: options,
+      auth_options: %{
```

Comment:
I would prefer it would be something like `nil | :basic | {:digest, options}`. That way you would not store inapplicable options. The type is the main property that is matched on. In that case, there would be no need to store ream and type if they are not used (as far as I understood they are not used in basic and no auth).